sssd_test_framework/utils/smartcard.py

```diff
@@ -0,0 +1,203 @@
+        :type cert_id: str, optional
+        :param pin: User PIN, defaults to "123456"
+        :type pin: str, optional
+        """
+        args: CLIBuilderArgs = {
+            "module": (self.cli.option.VALUE, "/usr/lib64/pkcs11/libsofthsm2.so"),
+            "login": (self.cli.option.SWITCH, True),
+            "pin": (self.cli.option.VALUE, pin),
+            "write-object": (self.cli.option.VALUE, cert_path),
+            "type": (self.cli.option.VALUE, "cert"),
+            "id": (self.cli.option.VALUE, cert_id),
+        }
+        command_line = self.cli.command("pkcs11-tool", args)
+        self.host.conn.run(command_line, env={"SOFTHSM2_CONF": self.SOFTHSM2_CONF_PATH})
+
+    def generate_self_signed_cert(
+        self,
+        key_path: str = "/tmp/selfsigned.key",
+        cert_path: str = "/tmp/selfsigned.crt",
+        subj: str = "/CN=Test Cert",
+    ) -> tuple[str, str]:
+        """
+        Generates a self-signed certificate and private key.
+
+        :param key_path: Output path for the private key, defaults to "/tmp/selfsigned.key"
+        :type key_path: str, optional
+        :param cert_path: Output path for the certificate, defaults to "/tmp/selfsigned.crt"
+        :type cert_path: str, optional
+        :param subj: Subject for the certificate, defaults to "/CN=Test Cert"
+        :type subj: str, optional
+        :return: Tuple of (key_path, cert_path)
+        :rtype: tuple
+        """
+        args: CLIBuilderArgs = {
+            "x509": (self.cli.option.SWITCH, True),
+            "nodes": (self.cli.option.SWITCH, True),
+            "sha256": (self.cli.option.SWITCH, True),
+            "days": (self.cli.option.VALUE, "365"),
+            "newkey": (self.cli.option.VALUE, "rsa:2048"),
+            "keyout": (self.cli.option.VALUE, key_path),
+            "out": (self.cli.option.VALUE, cert_path),
+            "subj": (self.cli.option.VALUE, subj),
+        }
+        command_line = self.cli.command("openssl req", args)
+        self.host.conn.run(command_line)
+        return key_path, cert_path
+
+    def insert_card(self) -> None:
+        """
+        Simulates card insertion by starting the smart card service.
+        """
+        self.svc.start("virt_cacard.service")
+
+    def remove_card(self) -> None:
+        """
+        Simulates card removal by stopping the smart card service.
+        """
+        self.svc.stop("virt_cacard.service")
+
+    def reset_service(self) -> None:
+        """
+        Restarts the virtual smart card service.
+        """
+        self.svc.restart("virt_cacard.service")
+
+    def setup_local_card(self, client: Client, username: str) -> None:
+        """
+        Setup local system for smart card authentication
+
+        .. code-block:: python
+            :caption: Example usage
+
+            @pytest.mark.topology(KnownTopology.Client)
+            def test_example(client: Client):
+                client.smart_card.setup_local_card(client, 'localuser1')
```

Comment:
@spoore1, you're right, it does look strange. 

```client.smart_card```, can we make an executive decision and just do ```client.smartcard``` for code and ```smart card``` in docs?